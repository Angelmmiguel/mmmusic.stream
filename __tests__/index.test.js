/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';

// Channels
const channels = require('./data/channels.test.json').channels;

import { Index } from '../pages/index.js';

describe('Index', () => {
  it('Basic rendering', () => {
    const app = shallow(<Index channels={ channels } filter={ 'All' } volume={ 50 }/>);
  });
});

