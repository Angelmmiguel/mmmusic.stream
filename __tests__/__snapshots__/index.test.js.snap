// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Basic rendering (snapshot) 1`] = `
<div
  className="jsx-4058071979"
>
  <div
    className="jsx-4058071979"
  >
    <header
      className="jsx-1783182476 Header"
    >
      <div
        className="jsx-1783182476 grid space-between align-center"
      >
        <div
          className="jsx-1783182476 cell"
        >
          <img
            alt="mmmusic logo"
            className="jsx-1783182476 Header__Logo"
            src="/static/logo.png"
          />
        </div>
        <div
          className="jsx-1783182476 cell Header__Current"
        >
          <section
            className="jsx-900726272 CurrentChannel "
          >
            <p
              className="jsx-900726272 CurrentChannel__NoChannel"
            >
              Click on a channel or press space to start the music
            </p>
          </section>
        </div>
        <div
          className="jsx-1783182476 cell Header__Volume"
        >
          <div
            className="jsx-1783182476 Header__Volume__Wrapper"
          >
            <div
              className="jsx-2829083726 Shortcuts"
            >
              <button
                className="jsx-2829083726"
                onClick={[Function]}
              >
                <svg
                  className="Shortcuts__Icon"
                  height="17"
                  viewBox="0 0 26 17"
                  width="2em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    keyboard
                  </title>
                  <g
                    fill="none"
                    fillRule="evenodd"
                  >
                    <rect
                      height="15"
                      rx="3"
                      stroke="#9B9898"
                      strokeWidth="2"
                      width="24"
                      x="1"
                      y="1"
                    />
                    <path
                      d="M3 11h3v3H3zm4 0h12v3H7zm13 0h3v3h-3zM3 7h3v3H3zm16 0h4v3h-4zm-4 0h3v3h-3zm-4 0h3v3h-3zM7 7h3v3H7zM3 3h3v3H3zm16 0h4v3h-4zm-4 0h3v3h-3zm-4 0h3v3h-3zM7 3h3v3H7z"
                      fill="#9B9898"
                    />
                  </g>
                </svg>
              </button>
            </div>
            <div
              className="jsx-407048431 Volume"
            >
              <svg
                className="feather feather-volume-1"
                fill="none"
                height="24"
                stroke="#F2302D"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 5L6 9H2v6h4l5 4V5zM15.54 8.46a5 5 0 0 1 0 7.07"
                />
              </svg>
              <div
                className="jsx-407048431 Volume__Slider"
              >
                <div
                  aria-disabled={false}
                  className="input-range"
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                >
                  <span
                    className="input-range__label input-range__label--min"
                  >
                    <span
                      className="input-range__label-container"
                    >
                      0
                    </span>
                  </span>
                  <div
                    className="input-range__track input-range__track--background"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                  >
                    <div
                      className="input-range__track input-range__track--active"
                      style={
                        Object {
                          "left": "0%",
                          "width": "50%",
                        }
                      }
                    />
                    <span
                      className="input-range__slider-container"
                      style={
                        Object {
                          "left": "50%",
                          "position": "absolute",
                        }
                      }
                    >
                      <span
                        className="input-range__label input-range__label--value"
                      >
                        <span
                          className="input-range__label-container"
                        >
                          50
                        </span>
                      </span>
                      <div
                        aria-controls={undefined}
                        aria-labelledby={undefined}
                        aria-valuemax={100}
                        aria-valuemin={0}
                        aria-valuenow={50}
                        className="input-range__slider"
                        draggable="false"
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                        role="slider"
                        tabIndex="0"
                      />
                    </span>
                  </div>
                  <span
                    className="input-range__label input-range__label--max"
                  >
                    <span
                      className="input-range__label-container"
                    >
                      100
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main>
      <section
        className="jsx-159405312 Genres"
      >
        <h1
          className="jsx-159405312 title"
        >
          Genres
        </h1>
        <section
          className="jsx-159405312 Genres__Selector"
        >
          <select
            className="jsx-159405312"
            defaultValue="All"
            onChange={[Function]}
          >
            <option
              className="jsx-159405312"
              value="All"
            >
              All
            </option>
            <option
              className="jsx-159405312"
              value="ambient"
            >
              ambient
            </option>
            <option
              className="jsx-159405312"
              value="chill out"
            >
              chill out
            </option>
            <option
              className="jsx-159405312"
              value="jazz"
            >
              jazz
            </option>
          </select>
        </section>
        <section
          className="jsx-159405312 Genres__Options"
        >
          <a
            className="jsx-447035556 active"
            href="/"
            onClick={[Function]}
          >
            All
          </a>
          <a
            className="jsx-447035556 "
            href="/genre/ambient"
            onClick={[Function]}
          >
            ambient
          </a>
          <a
            className="jsx-447035556 "
            href="/genre/chill-out"
            onClick={[Function]}
          >
            chill out
          </a>
          <a
            className="jsx-447035556 "
            href="/genre/jazz"
            onClick={[Function]}
          >
            jazz
          </a>
        </section>
      </section>
      <section
        className="jsx-2210179171 Channels"
      >
        <h1
          className="jsx-2210179171 title"
        >
          Channels (
          2
          )
        </h1>
        <section
          className="jsx-2210179171 grid small-grid--1of2 large-grid--1of4 grid--full wrap"
        >
          <article
            className="jsx-4159610607 Channel cell"
          >
            <section
              className="jsx-4159610607 Channel__Player"
            >
              <div
                className="Channel__Player__Thumbnail"
                style={
                  Object {
                    "backgroundImage": "url(https://i1.ytimg.com/vi/iu21hkjasdn/mqdefault.jpg)",
                  }
                }
              >
                <div
                  className="Channel__Player__Thumbnail__Play"
                >
                  <svg
                    alt="Play icon. Click to start the music"
                    className="feather feather-play-circle"
                    fill="none"
                    height="24"
                    stroke="#fff"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <path
                      d="M10 8l6 4-6 4V8z"
                    />
                  </svg>
                </div>
              </div>
              <button
                aria-label="Play the \\"Video test 1\\" channel."
                className="jsx-4159610607 Channel__Player__Button"
                onClick={[Function]}
              />
            </section>
            <h3
              className="jsx-4159610607"
            >
              Video test 1
            </h3>
          </article>
          <article
            className="jsx-4159610607 Channel cell"
          >
            <section
              className="jsx-4159610607 Channel__Player"
            >
              <div
                className="Channel__Player__Thumbnail"
                style={
                  Object {
                    "backgroundImage": "url(https://i1.ytimg.com/vi/ioqh2ojkndams/mqdefault.jpg)",
                  }
                }
              >
                <div
                  className="Channel__Player__Thumbnail__Play"
                >
                  <svg
                    alt="Play icon. Click to start the music"
                    className="feather feather-play-circle"
                    fill="none"
                    height="24"
                    stroke="#fff"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <path
                      d="M10 8l6 4-6 4V8z"
                    />
                  </svg>
                </div>
              </div>
              <button
                aria-label="Play the \\"Video test 2\\" channel."
                className="jsx-4159610607 Channel__Player__Button"
                onClick={[Function]}
              />
            </section>
            <h3
              className="jsx-4159610607"
            >
              Video test 2
            </h3>
          </article>
        </section>
      </section>
    </main>
    <footer
      className="jsx-1306090244"
    >
      <section
        className="jsx-1306090244"
      >
        <div
          className="jsx-1306090244 grid"
        >
          <div
            className="jsx-1306090244 cell"
          >
            © mmmusic 
            2018
            . An
             
            <a
              className="jsx-1306090244"
              href="https://github.com/Angelmmiguel/mmmusic.stream"
              rel="noopener"
              target="_blank"
            >
              Open Source project
            </a>
             
            created with
             
            <a
              className="jsx-1306090244"
              href="https://ko-fi.com/angelmmiguel"
              rel="noopener"
              target="_blank"
            >
              <svg
                className="jsx-1306090244"
                fill="none"
                height="16"
                stroke="currentColor"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  className="jsx-1306090244"
                  d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
                />
              </svg>
            </a>
             
            by
             
            <a
              className="jsx-1306090244"
              href="https://twitter.com/_angelmm"
              rel="noopener"
              target="_blank"
            >
              Angel
            </a>
          </div>
          <div
            className="jsx-1306090244 cell tr"
          >
            mmmusic is not affiliated with
             
            <a
              className="jsx-1306090244"
              href="https://youtube.com"
              rel="noopener"
              target="_blank"
            >
              Youtube
            </a>
          </div>
        </div>
      </section>
    </footer>
  </div>
</div>
`;
